src/main/daml/Daml/Finance/Instrument/Generic/Util.daml

```diff
@@ -62,6 +62,28 @@ createFixRatePaymentClaims schedule periodicSchedule useAdjustedDatesForDcf coup
   prepareAndTagClaims couponClaims "Fix rate payment"
 -- FIXED_RATE_BOND_COUPON_CLAIMS_END
 
+-- | Calculate a fix rate amount (if a credit event has not yet happened) for each payment date and create claims.
+createConditionalCreditFixRatePaymentClaims : Applicative f => [SchedulePeriod] -> PeriodicSchedule -> Bool -> Decimal -> Decimal -> DayCountConventionEnum -> Deliverable -> Observable -> f [TaggedClaim]
```

Comment:
Is there a reason this has to be monadic, and cannot just be a pure function? Also, it doesn't seem to need to return a list, but could just return `TaggedClaim`.

Reply:
You are right, I have changed it to a pure function that only returns `TaggedClaim` (not a list).